lsa-mis/room_ready
# Lra 908 room ready rover interface form for common attributes

Changes:
- Added Rover Form for Common Attribute States
- Added validations
  - if there is a checkbox option, one option needs to be selected
  - if there is a quantity box, a number needs to be added
- Added checks for the page (tentative). Cannot be accessed if:
  - room doesnt exist
  - room state doesnt exist for room for today (this is assumed/prerequisite since the step before this is saving RoomState)
  - if Common Attribute States have already been saved for today
- Interface looks nice on mobile
- main link is `/common_attribute_states/new?room_id=<ROOM_ID>`

To test:
1. Pick a room, go to its #show page
2. Click on the 'Edit Common Attribute States' link (note that this link is temporary)
3. This will throw an error: room_state needs to be created
4. In the rails console, run: `Room.find(<ROOM_ID>).room_states.create` (to create a RoomState for that specific room)
5. Click on the link again. It should take you to the form
6. Fill the whole form (validations are present, so everything needs to be filled/selected)
7. Save
8. Takes you back to the room, and common_attributes are created (changes are reflected in the database)
9. Clicking on the link again will throw an error, since you have already saved Common Attribute States for today
